frontend/src/metabase/query_builder/components/filters/modals/BulkFilterModal/BulkFilterModal.tsx

```diff
@@ -77,15 +94,17 @@ const BulkFilterModal = ({
     <ModalRoot>
       <ModalHeader>
         <ModalTitle>{getTitle(question, query)}</ModalTitle>
+        <FieldSearch value={searchQuery} onChange={setSearchQuery} />
         <ModalCloseButton onClick={onClose}>
           <Icon name="close" />
         </ModalCloseButton>
       </ModalHeader>
-      {sections.length === 1 ? (
+      {sections.length === 1 || searchItems ? (
         <BulkFilterModalSection
           query={query}
           filters={filters}
-          section={sections[0]}
+          items={searchItems ?? sections[0].items}
+          isSearch={!!searchItems}
```

Comment:
we need to pass down a prop indicating whether this is a search result so that the subcomponent can render the table name in addition to the field name